test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/CachedExpressionCompilerTest.cs

```diff
@@ -25,6 +25,21 @@ public void Process_IdentityExpression()
             Assert.Same(model, result);
         }
 
+        [Fact]
+        public void Process_CachesIdentityExpression()
```

Comment:
These new "Caches" tests only cover reference equality of the passed expressions. Should succeed if `GetTestModelExpression(...)` were called twice. That's the test I was hoping for.